homeassistant/components/slack/notify.py

```diff
@@ -74,16 +82,55 @@
 )
 
 
-async def async_get_service(hass, config, discovery_info=None):
+class AuthDictT(TypedDict, total=False):
+    """Type for auth request data."""
+
+    auth: BasicAuth
+
+
+class FormDataT(TypedDict):
+    """Type for form data, file upload."""
+
+    channels: str
+    filename: str
+    initial_comment: str
+    title: str
+    token: str
+
+
+class MessageT(TypedDict, total=False):
+    """Type for message data."""
+
+    link_names: bool
+    text: str
+    username: str  # Optional
+    icon_url: str  # Optional
+    icon_emoji: str  # Optional
+    blocks: List[Any]  # Optional
+
+
+async def async_get_service(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+) -> Optional[SlackNotificationService]:
     """Set up the Slack notification service."""
-    session = aiohttp_client.async_get_clientsession(hass)
-    client = WebClient(token=config[CONF_API_KEY], run_async=True, session=session)
+    session: ClientSession = aiohttp_client.async_get_clientsession(hass)
```

Comment:
Is the `ClientSession` annotation necessary? I was under the impression that the type could be inferred from `aiohttp_client.async_get_clientsession(hass)`.

I'm open to your thoughts, but we should be consistent: either variables get annotations (I notice places below that don't) or they don't.

Reply:
If the called function is typed, it's not necessary. I tend to add it from time to time, but especially when a function from a third party library is called, to help the type checker and autocompletion a bit.

I'll remove the variable annotations, they are not necessary here.